tests/spec.rs

```diff
@@ -1,136 +1,66 @@
-use std::io::Read;
-
-use std::{fs::File, io::Write};
-use wabt::script::{Command, CommandKind, ScriptParser};
+use std::process::Command;
 use wasker::compiler;
 
-fn _run_spec_test(testname: &str) {
-    let path = "./tests/spec/test/core";
-
-    // read wast
-    let mut wat = String::new();
-    let wat_path = std::path::Path::new(path).join(format!("{testname}.wast"));
-    println!("open file: {wat_path:?}");
-    let mut file = std::fs::File::open(wat_path).expect("error open file");
-    file.read_to_string(&mut wat).expect("cannot read file");
-
-    // parse
-    let mut parser = ScriptParser::<f32, f64>::from_str(&wat).expect("error get parser");
-    while let Some(Command { kind, .. }) = parser.next().expect("cannot parse next") {
-        match kind {
-            CommandKind::Module { module, name } => {
-                // Convert the module into the binary representation and check the magic number.
-                let module_binary = module.into_vec();
-                assert_eq!(&module_binary[0..4], &[0, 97, 115, 109]);
-
-                // Output wasm for debug
-                let mut file = File::create("tmp.wasm").expect("cannot open file");
-                file.write_all(&module_binary)
-                    .expect("fail to write binary");
-                file.flush().expect("fail to flush");
+fn ensure_log_dir(log_dir: &str) {
+    if !std::path::Path::new(log_dir).exists() {
+        std::fs::create_dir_all(log_dir).expect("Failed to create log directory");
+    }
+}
 
-                // Compile test
-                println!("Compile Module {name:?}");
-                let args = compiler::Args {
-                    input_file: "tmp.wasm".into(),
-                    output_file: "/tmp/wasm.o".into(),
-                };
-                compiler::compile_wasm(&module_binary, &args).expect("compile failed");
-            }
-            _ => {
-                // TODO: support assertion
-            }
-        }
+fn compile_for_executable(output_path: &str, wasm_path: &str, wasi_wrapper_path: &str) {
+    let compile_status = Command::new("gcc")
+        .args(["-o", output_path, wasm_path, wasi_wrapper_path, "-no-pie"])
+        .status()
+        .expect("Failed to compile with GCC");
+    if !compile_status.success() {
+        panic!("GCC compilation failed");
     }
 }
 
-macro_rules! _spec_test {
-    ($testname:ident, $fname:expr) => {
-        #[test]
-        #[allow(non_snake_case)]
-        fn $testname() {
-            run_spec_test($fname);
-        }
+fn run_test(testcase: &str) {
+    let project_root = env!("CARGO_MANIFEST_DIR");
+    let log_dir = format!("{project_root}/target/test_logs");
+    ensure_log_dir(&log_dir);
+
+    let wat_path = format!("{project_root}/tests/wat/{testcase}.wat");
+    let wasker_output_path = format!("{log_dir}/wasm.o");
+    let executable_path = format!("{log_dir}/exec.out");
```

Comment:
This might cause conflict between test cases when running them concurrently. We would need to name artifact files differently for each case.